esp/smc/SMCLib/TpWrapper.cpp

```diff
@@ -457,9 +456,16 @@ void CTpWrapper::getTpEspServers(IArrayOf<IConstTpEspServer>& list)
                 StringBuffer xpath;
                 xpath.appendf("EspService[@name='%s']", service);
                 IPropertyTree* pServiceNode = root->queryPropTree(xpath.str());
-                const char* serviceType = pServiceNode ? pServiceNode->queryProp("Properties/@type") : "INVALID";
-                pTpBinding->setServiceType(eqEsp);
+                const char* serviceType = pServiceNode ? pServiceNode->queryProp("Properties/@type") : NULL;
+                if (serviceType && *serviceType)
+                    pTpBinding->setServiceType(serviceType);
 
+                if (pServiceNode && (clientVersion >= 1.23))
```

Comment:
You don't have to pass in or check the version for setting fields unless there is a lot of overhead in getting, transforming or setting the information.  The interface will handle the versioning for you.  Checking the version is only an optimization.